src/seam-connect/client.ts

```diff
@@ -137,78 +93,121 @@ export class Seam extends Routes {
   public async makeRequest<T>(
     request: AxiosRequestConfig
   ): Promise<SuccessfulAPIResponse<T>> {
-    try {
-      const response = await this.client.request(request)
-      return response.data
-    } catch (error) {
-      if (axios.isAxiosError(error) && error.response) {
-        if (error.response.data.error?.type === "invalid_input") {
-          throw new SeamMalformedInputError(
-            error.response.data.error.validation_errors
-          )
+    return makeRequest<T>(this.client, request)
+  }
+
+  static async getClientSessionToken(options: {
+    publishableKey?: string
+    userIdentifierKey: string
+    endpoint?: string
+    workspaceId?: string
+    apiKey?: string
+  }): Promise<APIResponse<ClientSessionResponse>> {
+    const { apiKey, endpoint, axiosOptions } =
+      getSeamClientOptionsWithDefaults(options)
+
+    if (!options.userIdentifierKey) {
+      throw new Error("userIdentifierKey is required")
+    }
+
+    const getKeyHeaders = (): AxiosRequestHeaders => {
+      const { publishableKey } = options
+      if (publishableKey) {
+        if (!publishableKey.startsWith("seam_pk")) {
+          throw new Error("Invalid publishableKey")
         }
+        return { "seam-publishable-key": publishableKey }
+      }
 
-        throw new SeamAPIError(
-          error.response.status,
-          error.response.headers["seam-request-id"],
-          (error.response.data as ErroredAPIResponse).error
-        )
+      if (apiKey) {
+        if (!apiKey?.startsWith("seam_")) {
+          throw new Error("Invalid apiKey")
+        }
+        return { "seam-api-key": apiKey }
       }
 
-      throw error
+      throw new Error("Must provide a publishableKey or apiKey")
     }
-  }
 
-  static async getClientSessionToken(
-    ops: CSTParams
-  ): Promise<APIResponse<ClientSessionResponse>> {
-    const { apiKey, endpoint, axiosOptions } =
-      getSeamClientOptionsWithDefaults(ops)
-    let headers: AxiosRequestHeaders = {
+    const headers = {
+      "seam-user-identifier-key": options.userIdentifierKey,
```

Comment:
not opposed to this header, but i think i lean towards this being preferred in the payload, allowed in header, headers distract from the concept that we're basically an RPC API, we only use them where convention or security pushes us to. That being said, I think it's ok that it's accepted as a header.

Reply:
I think this is the existing implementation (not added in this PR). We can make the switch but will need to check the endpoint first to see if it will accept this in the body.

Reply:
yep that's fine!!!